libnetdata/libnetdata.c

```diff
@@ -1492,3 +1492,37 @@ char *read_by_filename(char *filename, long *file_size)
         *file_size = size;
     return contents;
 }
+
+char *find_and_replace(const char *src, const char *find, const char *replace, const char *where)
+{
+   size_t size        = strlen(src) + 1;
+   size_t find_len    = strlen(find);
+   size_t repl_len    = strlen(replace);
+   char *value, *dst, *temp;
+
+   value = mallocz(size);
+   dst = value;
+   if ( likely(where) ) {
+
+       size_t count = where - src;
+
+       size += repl_len - find_len;
+
+       temp = reallocz(value, size);
```

Comment:
Instead to call `realloc` and iterate with kernel one more time, I will give another algorithm:

1 - Measure the `src size`
2 - Measure the `find size`.
3 - Measure the `replace size`.
4 - Calculate: `src size` - (`replace size` - `find size`) + 1
5  - Allocate the calculated result of `4`.

Reply:
You are right of course, there's no need for the realloc, if we can find the size before the malloc. Changed with https://github.com/netdata/netdata/pull/10961/commits/a5eab8a81f68e87c1d00b58559d8e43f85d36b3e